staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -101,55 +68,100 @@ func WithAudit(handler http.Handler, requestContextMapper request.RequestContext
 			responsewriters.InternalError(w, req, errors.New("no context found for request"))
 			return
 		}
-		attribs, err := GetAuthorizerAttributes(ctx)
-		if err != nil {
-			responsewriters.InternalError(w, req, err)
-			return
+
+		// prefer the id from the headers. If not available, create a new one.
+		// TODO(security): do we want to forbid the header for non-front-proxy users?
+		var id string
+		ids := req.Header[AuditIDHeader]
```

Comment:
I'm not convinced we need the ID chain. Wouldn't the IP chain be sufficient to get this information?